### ISSUE TITLE
AST scenario object not available in Before and after

### ISSUE THREAD

**Issue by nikulkarni** [cucumber/cucumber-js#289]
I'm using cucumber@0.4.7 and trying to use [AST scenario][1] object methods like `getSteps()` in either `Before` or `After`, however It seems that I have [API scenario][2] object available instead of [AST scenario][1] object since it prints all methods available in [API scenario][2] object. Here is the code below,

```
 this.After(function (scenario, callback) {
        console.dir(scenario);
});
```
This prints below
```
{ getKeyword: [Function: getKeyword],
  getName: [Function: getName],
  getDescription: [Function: getDescription],
  getUri: [Function: getUri],
  getLine: [Function: getLine],
  getTags: [Function: getTags],
  isSuccessful: [Function: isSuccessful],
  isFailed: [Function: isFailed],
  isPending: [Function: isPending],
  isUndefined: [Function: isUndefined],
  getAttachments: [Function: getAttachments],
  attach: [Function: attach] }
```
I thought of constructing the object manually, but it's not outputting expected values. For example,
```
 this.After(function (scenario, callback) {
        var ast_scenario = require('cucumber/lib/cucumber/ast/scenario');
        var scene = ast_scenario((scenario.getKeyword(),scenario.getName(),scenario.getDescription(),scenario.getUri(),scenario.getLine()));
        console.log(scene.getSteps().length()); //this prints 0 for some reason
});
```

Am I missing anything here? 

Thanks!

[1]: https://github.com/cucumber/cucumber-js/blob/master/lib/cucumber/ast/scenario.js "AST scenario"
[2]: https://github.com/cucumber/cucumber-js/blob/master/lib/cucumber/api/scenario.js "API scenario"

**Comment by h2non** [comment]
The same is applied for the `Before` hook. The documentation is incongruent.

```js
this.Before(function (scenario, callback) {
  console.log(scenario.getName(), "(" + scenario.getUri() + ":" + scenario.getLine() + ")");
  callback();
});
```

Scenario should implement the [AST Scenario](https://github.com/cucumber/cucumber-js/blob/master/lib/cucumber/ast/scenario.js) according to the docs, but it doesn't

**Comment by ruslango** [comment]
+1

**Comment by bristly** [comment]
Hi guys,
Our team faced with the same issue.
We should create an additional step dynamically.
And we try this:

``` javascript
var step = new Cucumber.Ast.Step(Cucumber.Ast.Step.OUTCOME_STEP_KEYWORD, 
"Name of a step", scenario.getUri(), scenario.getSteps().length() + 1);
scenario.addStep(step);
```
Or maybe you know else way to create a new step?

**Comment by sJhonny-e** [comment]
+1 this is super annoying

**Comment by charlierudolph** [comment]
Fixed in 13506320c2ae9e210b122b5e423f09815a1668a1. The README now correctly states that the scenario is Cucumber.Api.Scenario. The Ast object has methods that allow the method to be changed so it isn't exposed. If you have specific methods you want added to the Api object, please open another issue.
